providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/objects.h>
+#include <openssl/params.h>
+#include "internal/param_build.h"
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+
+static OSSL_OP_keymgmt_importdomparams_fn ec_importdomparams;
+static OSSL_OP_keymgmt_exportdomparams_fn ec_exportdomparams;
+static OSSL_OP_keymgmt_importkey_fn ec_importkey;
+static OSSL_OP_keymgmt_exportkey_fn ec_exportkey;
+
+static ossl_inline
+EC_GROUP *params_to_domparams(const OSSL_PARAM params[])
+{
+    const OSSL_PARAM *param_ec_name;
+    EC_GROUP *ecg = NULL;
+    char *curve_name = NULL;
+    int ok = 0;
+
+    param_ec_name = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_EC_NAME);
+    if (param_ec_name == NULL) {
+        /* explicit parameters */
+
+        /*
+         * TODO(3.0): should we support explicit parameters curves?
+         */
+        return NULL;
+    } else {
+        /* named curve */
+        int curve_nid;
+
+        if (!OSSL_PARAM_get_utf8_string(param_ec_name, &curve_name, 0)
+                || curve_name == NULL
+                || (curve_nid = OBJ_sn2nid(curve_name)) == NID_undef)
```

Comment:
Side note: We're going to need to do something else when we move this to the FIPS provider, because `OBJ_sn2nid` doesn't exist there.

Reply:
Can the scripts generating the object table be updated to generate a namemap sn2nid of NIDs supported by the FIPS module?

The EC module as it is now heavily depends on NIDs, so we need some kind of mechanism to go from nid2string and viceversa inside the FIPS EC code.

Reply:
There's only a very limited number of curve names that we care about for the FIPS module so I'm sure we can come up with something. But not for this PR...

Reply:
EC_curve_nist2nid() ?

Reply:
> EC_curve_nist2nid() ?

@slontis I totally forgot about it!! Good point!

I am looking at tit again: what about aliases?
In theory we would want that to map foe example

"secp256r1", "prime256v1" or "P-256" to `NID_X9_62_prime256v1`

at the very least we need "prime256v1" as that is the result we get from `OBJ_nid2sn()`

Reply:
Anyway, the FIPS adjustments should be handled in a separate PR after this is in!

Reply:
We have a bunch of FIPS adjustments getting accumulated.  We still don't have a single PKEY algorithm in there...  :wink: